src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class BookControllerTest {
+
+
+    @Autowired
+    private MockMvc mvc;
+    @MockBean
+    private BookRepository bookRepository;
+
+    private Book book;
+
+    @BeforeEach
+    public void setup() {
+        book = new Book("comedian", "author", "image", "comedyTest", "subtitle", "norma", "2020", "pages", "isbn");
+
+    }
+
+    @Test
+    public void whenAddBook_ThenCreatedOk() throws Exception {
+
+        String bookBody = "{\"genre\":\"drama\",\"author\":\"author\",\"image\":\"image\",\"title\":\"book3\",\"subtitle\":\"subtitle\",\"publisher\":\"publisher\",\"year\":\"2020\",\"pages\":\"pages\",\"isbn\":\"isbn\"}";
+        when(bookRepository.save(book)).thenReturn(book);
+
+        mvc.perform(post("/books")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(bookBody))
+                .andExpect(status().isCreated());
+
+    }
+
+    @Test
+    public void whenFindByTitle_ThenFoundOk() throws Exception {
+
+        String bookPath = "comedyTest";
+        Optional<Book> optionalBook = Optional.of(book);
+
+        when(bookRepository.findByTitle(bookPath)).thenReturn(optionalBook);
+
+        mvc.perform(get("/books/title/{bookTitle}", "comedyTest"))
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title").value("comedyTest"));
+
+        //Assertions.assertEquals("application/json", result.getResponse().getContentType());
+        //Assertions.assertEquals();
+
+    }
+
+    @Test
+    public void whenDeleteById_ThenDeletedOk() throws Exception {
+
+        book.setId(1);
+        Optional<Book> optionalBook = Optional.of(book);
+
+        when(bookRepository.findById(new Long(1))).thenReturn(optionalBook);
+        doNothing().when(bookRepository).deleteById(any(Long.class));
+
+        mvc.perform(delete("/books/{id}", 1))
+                .andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void whenUpdateBook_ThenUpdatedOk() throws Exception {
+
+        book.setId(1);
+        Optional<Book> optionalBook = Optional.of(book);
+        String bookBody = "{\"id\":\"1\",\"genre\":\"drama\",\"author\":\"author\",\"image\":\"image\",\"title\":\"book3\",\"subtitle\":\"subtitle\",\"publisher\":\"publisher\",\"year\":\"2020\",\"pages\":\"pages\",\"isbn\":\"isbn\"}";
```

Comment:
Hello! It's ok, but it's easier to use the ObjectMapper.writeValueAsString(object) to build a JsonString. 😁

Reply:
it was resolved at last commit